### PR TITLE
ss-175: Add a toaster to hide menu items in the header if the screen resolution is small

### PR DESCRIPTION
**Pull request by teslmik**
**to make smooth scrolling work in chrome: chrome://flags/#smooth-scrolling** 

cAscmk1qaS

### CODE REVIEW

File: frontend/src/bundles/common/components/header/styles.module.scss [hunk 1]
```diff
@@ -73,15 +80,42 @@
     color: var(--color-blue-600);
 }
 
+@media (max-width: 992px) {
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/header/styles.module.scss:83]: Use variables

File: frontend/src/bundles/common/helpers/handle-scroll/handle-scroll.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+            const isLastLinkVisibleAtBottom =
+                isLastLink &&
+                window.innerHeight + window.pageYOffset >=
+                    document.body.offsetHeight;
+
+            const isElementVisible =
+                isLastLinkVisibleAtBottom || elementTop < window.pageYOffset;
+
+            if (isElementVisible) {
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/helpers/handle-scroll/handle-scroll.ts:28]: Avoid nesting if statements

File: frontend/src/bundles/common/helpers/handle-scroll/handle-scroll.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                setState(link.to);
+                isLastLinkVisible =
+                    link.to === menuLinks[menuLinks.length - 1].to ||
+                    isLastLinkVisible;
+            }
+        }
+
+        if (
+            !isLastLinkVisible &&
+            window.innerHeight + window.scrollY >= document.body.offsetHeight
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/helpers/handle-scroll/handle-scroll.ts:38]: Move this comparison to a variable

File: frontend/src/bundles/common/pages/budgets/budgets.tsx [hunk 4]
```diff
@@ -50,17 +53,25 @@ const Budgets: React.FC<Properties> = ({ budgetCards }) => {
                                 date={card.date}
                             />
                         ))}
-                        <Button
+                        <div
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/pages/budgets/budgets.tsx:56]: Why 'button' is replaced by 'div'?

File: frontend/src/bundles/common/components/scroll-nav/scroll-nav.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    className,
+}) => {
+    const linkReference = useRef<HTMLDivElement>(null);
+
+    const handleClick = useCallback((): void => {
+        const element: HTMLElement | null = document.querySelector(scrollToId);
+
+        if (element) {
+            const elementTop =
+                element.getBoundingClientRect().top + window.pageYOffset - 80;
```
[the remark this answers is not in the document]
Comment (teslmik) [reply frontend/src/bundles/common/components/scroll-nav/scroll-nav.tsx:21]: height heder = 80px.
header position: fixed

File: frontend/src/bundles/common/pages/budgets/budgets.tsx (hunk 4 above)
Comment (teslmik) [reply frontend/src/bundles/common/pages/budgets/budgets.tsx:56]: there was an error: button cannot be a button's parent

File: frontend/src/bundles/common/components/header/styles.module.scss (hunk 1 above)
Comment (teslmik) [reply frontend/src/bundles/common/components/header/styles.module.scss:83]: fixed in merge with dev

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**